src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -165,155 +263,714 @@ private MessageConsumer newReplyStream(
+
+            OctetsFW openWithFrameHeader = payloadRW.wrap(writeBuffer, DataFW.FIELD_OFFSET_PAYLOAD, writeBuffer.capacity())
+                .put(updatedFrameHeader.buffer(), 0, updatedFrameHeader.sizeof())
+                .put(open.buffer(), offsetOpenFrame, open.sizeof())
+                .build();
+
+            doNetworkData(traceId, authorization, 0L, openWithFrameHeader);
         }
 
-        private void doReset(
-            long traceId)
+        private void doEncodeBegin(
+            long traceId,
+            long authorization)
         {
-            final ResetFW reset = resetRW.wrap(writeBuffer, 0, writeBuffer.capacity())
-                .routeId(routeId)
-                .streamId(initialId)
-                .traceId(traceId)
+            int offsetBeginFrame = amqpFrameHeaderRW.limit();
+
+            final AmqpBeginFW begin = amqpBeginRW
+                .wrap(frameBuffer, offsetBeginFrame, frameBuffer.capacity())
+                .remoteChannel(channelId)
+                .nextOutgoingId(nextTransferId)
+                .incomingWindow(writeBuffer.capacity())
+                .outgoingWindow(outgoingWindow)
+                .build();
+
+            AmqpFrameHeaderFW originalFrameHeader = amqpFrameHeaderRW.build();
+
+            final AmqpFrameHeaderFW updatedFrameHeader =
+                amqpFrameHeaderRW.wrap(frameBuffer, 0, frameBuffer.capacity())
+                    .size(originalFrameHeader.sizeof() + begin.sizeof())
+                    .doff(originalFrameHeader.doff())
+                    .type(originalFrameHeader.type())
+                    .channel(originalFrameHeader.channel())
+                    .performative(b -> b.domainId(d -> d.set(originalFrameHeader.performative().domainId()))
+                        .descriptorId(d2 -> d2.set(originalFrameHeader.performative().descriptorId().get())))
+                    .build();
+
+            OctetsFW beginWithFrameHeader = payloadRW.wrap(writeBuffer, DataFW.FIELD_OFFSET_PAYLOAD, writeBuffer.capacity())
+                .put(updatedFrameHeader.buffer(), 0, updatedFrameHeader.sizeof())
+                .put(begin.buffer(), offsetBeginFrame, begin.sizeof())
                 .build();
 
-            receiver.accept(reset.typeId(), reset.buffer(), reset.offset(), reset.sizeof());
+            doNetworkData(traceId, authorization, 0L, beginWithFrameHeader);
         }
 
-        private void doWindow(
+        private void doEncodeAttach(
             long traceId,
-            int initialCredit)
+            long authorization,
+            AmqpRole role,
+            AmqpSenderSettleMode senderSettleMode,
+            AmqpReceiverSettleMode receiverSettleMode,
+            String address)
         {
-            if (initialCredit > 0)
+            int offsetAttachFrame = amqpFrameHeaderRW.limit();
+
+            AmqpAttachFW.Builder attachRW = amqpAttachRW.wrap(frameBuffer, offsetAttachFrame, frameBuffer.capacity())
+                .name(asStringFW(linkName))
```

Comment:
You can just pass `linkName` directly here without `asStringFW`, right?
If all similar method call sites can be removed, then please also remove `asStringFW` method.

Reply:
For `list` `Builder`, if a field is `variant`, the setter's argument should be the `variant`'s `oftype` type. And since the type of the field `name` is `AmqpString` (which is `variant`), we can only set as `StringFW` right now.

Reply:
Discussed and resolved on Slack. (Use a StringFW shared flyweight instance and populate the string)